Assets/Labrys/Editor/FeatureEditor/Panels/ToolBox.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Labrys.Editor.FeatureEditor.Panels
+{
+	[System.Serializable]
+	public class ToolBox : InternalPanel
+	{
+		[SerializeField]
+		private List<Tool> tools;
+		[SerializeField]
+		private int selectedTool;
+
+		public Tool ActiveTool { get { return tools[selectedTool]; } }
+
+		public ToolBox(EditorWindow window, DockPosition alignment, float scale) : base(window, alignment, scale)
+		{
+			tools = new List<Tool>();
+			selectedTool = 0;
+		}
+
+		public override void Draw()
+		{
+			base.Draw();
+			selectedTool = GUI.SelectionGrid(bounds, selectedTool, GetToolNames(), 1);
+		}
+
+		public override bool HandleEvent(Event e)
+		{
+			bool guiChanged = false;
+
+			switch(e.type)
+			{
+			case EventType.KeyDown:
+				if(e.isKey)
```

Comment:
can refactor this to a boolean check(for the KeyDown) + a switch statement (for the KeyCodes)